security/i2pd/Portfile

```diff
@@ -0,0 +1,153 @@
+        xinstall -d ${destroot}${prefix}/share/${name}
+        copy ${worksrcpath}/contrib/certificates ${destroot}${prefix}/share/${name}
+    }
+    xinstall -d ${destroot}${prefix}/share/doc/${name}
+    xinstall -W ${worksrcpath} -m 0644 ChangeLog LICENSE README.md ${destroot}${prefix}/share/doc/${name}
+    # backup conf files
+    foreach f {i2pd.conf subscriptions.txt tunnels.conf tunnels.d} {
+        if {[file exists ${prefix}/etc/${name}/${f}]} {
+            move ${destroot}${prefix}/etc/${name}/${f} ${destroot}${prefix}/etc/${name}/${f}.new
+            copy ${prefix}/etc/${name}/${f} ${destroot}${prefix}/etc/${name}/${f}.mp_backup
+            file attributes ${destroot}${prefix}/etc/${name}/${f}.mp_backup -owner ${i2puser} -group ${i2puser} -permissions 0664
+        }
+    }
+}
+
+notes "
+i2pd status can be checked on: http://127.0.0.1:7070
+You can edit i2pd settings in ${i2pddir}
+See also: https://i2pd.readthedocs.io/en/latest/user-guide/configuration
+Re using i2pd with web-browser: https://github.com/PurpleI2P/i2pdbrowser
+"
+
+post-activate {
+    # DataDirectory and PID file Ddirectory permissions
+    file attributes ${i2pddir} -owner ${i2puser} -group ${i2puser} -permissions 0755
+    file attributes ${prefix}/var/run/${name} -owner ${i2puser} -group ${i2puser} -permissions 0755
+
+    # restore conf files
+    foreach f {i2pd.conf subscriptions.txt tunnels.conf tunnels.d} {
+        if {![file exists ${prefix}/etc/${name}/${f}]} {
+            if {[file exists ${prefix}/etc/${name}/${f}.mp_backup]} {
+                copy ${prefix}/etc/${name}/${f}.mp_backup ${prefix}/etc/${name}/${f}
+            } else {
+                copy ${prefix}/etc/${name}/${f}.new ${prefix}/etc/${name}/${f}
+            }
+            file attributes ${prefix}/etc/${name}/${f} -owner ${i2puser} -group ${i2puser} -permissions 0664
+        }
+    }
+
+    if {![file exists ${i2pddir}]} {
+        xinstall -m 0664 -d ${i2pddir}
+    }
+    ln -sf ${prefix}/share/${name}/certificates ${i2pddir}/certificates
+    ln -sf ${prefix}/etc/${name}/tunnels.d ${i2pddir}/tunnels.d
+    ln -sf ${prefix}/etc/${name}/i2pd.conf ${i2pddir}/i2pd.conf
+    ln -sf ${prefix}/etc/${name}/subscriptions.txt ${i2pddir}/subscriptions.txt
+    ln -sf ${prefix}/etc/${name}/tunnels.conf ${i2pddir}/tunnels.conf
+}
+
+startupitem.create      yes
+# Option --service has no effect on macOS: https://i2pd.readthedocs.io/en/latest/user-guide/configuration
+startupitem.executable  sudo -u ${i2puser} ${prefix}/bin/i2pd \
+                        --datadir=${i2pddir} \
+                        --conf=${i2pddir}/i2pd.conf \
+                        --certsdir=${i2pddir}/certificates \
+                        --tunconf=${i2pddir}/tunnels.conf \
+                        --log=${prefix}/var/log/${name} \
+                        --bob.enabled=true \
+                        --i2cp.enabled=true \
+                        --daemon
```

Comment:
```suggestion
startupitem.create      yes
startupitem.user        ${i2puser}
startupitem.group       ${i2puser}
# Option --service has no effect on macOS: https://i2pd.readthedocs.io/en/latest/user-guide/configuration
startupitem.executable  ${prefix}/bin/i2pd \
                        --datadir=${i2pddir} \
                        --conf=${i2pddir}/i2pd.conf \
                        --certsdir=${i2pddir}/certificates \
                        --tunconf=${i2pddir}/tunnels.conf \
                        --log=${prefix}/var/log/${name} \
                        --bob.enabled=true \
                        --i2cp.enabled=true \
                        --daemon
startupitem.pidfile     auto ${prefix}/var/run/${name}/i2pd.pid
startupitem.logfile     ${destroot}${prefix}/var/log/${name}/daemon.log
startupitem.logevents   yes
```

I suggest to read "docs" for startupitem: https://github.com/macports/macports-base/blob/master/src/port1.0/portstartupitem.tcl :)

I not sure that I've pointed the right log file (it should be different one), and not sure about pid file and its behaviour, but you should get an idea :)

Reply:
I actually had those earlier, but decided to minimize the code. `pid` and `log` are specified in `conf` file. But let me try to add these back.

Reply:
@barracuda156 `pid` and `log` are specified in `i2pd` config, here I've specified them for `startupitem`. Log should allow to debug it (right now all logs from startupitem is going to `/dev/null`) and pid should fix unload issue (I hope)